molgenis-api-permissions/src/main/java/org/molgenis/api/permissions/PermissionsController.java

```diff
@@ -0,0 +1,473 @@
+import org.molgenis.data.security.permission.model.LabelledType;
+import org.molgenis.data.security.permission.model.Permission;
+import org.molgenis.security.acl.ObjectIdentityService;
+import org.molgenis.security.core.PermissionSet;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.acls.model.Sid;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Api("Permissions API")
+@RestController
+@RequestMapping(BASE_URI)
+@Transactional
+public class PermissionsController extends ApiController {
+
+  private static final String PERMISSION_API_IDENTIFIER = "permissions";
+  public static final String BASE_URI = "/api/" + PERMISSION_API_IDENTIFIER;
+  static final Integer DEFAULT_PAGE = 1;
+  static final Integer DEFAULT_PAGESIZE = 100;
+
+  private static final String TYPE = "type";
+  public static final String TYPES = TYPE + "s";
+  private static final String TYPE_ID = TYPE + "Id";
+
+  private static final String OBJECT = "object";
+  public static final String OBJECTS = OBJECT + "s";
+  private static final String OBJECT_ID = OBJECT + "Id";
+
+  private final PermissionService permissionService;
+  private final RSQLParser rsqlParser;
+  private final ObjectIdentityService objectIdentityService;
+  private final UserRoleTools userRoleTools;
+  private final EntityHelper entityHelper;
+
+  public PermissionsController(
+      PermissionService permissionService,
+      RSQLParser rsqlParser,
+      ObjectIdentityService objectIdentityService,
+      UserRoleTools userRoleTools,
+      EntityHelper entityHelper) {
+    super(PERMISSION_API_IDENTIFIER, 1);
+    this.permissionService = requireNonNull(permissionService);
+    this.rsqlParser = requireNonNull(rsqlParser);
+    this.objectIdentityService = requireNonNull(objectIdentityService);
+    this.userRoleTools = requireNonNull(userRoleTools);
+    this.entityHelper = requireNonNull(entityHelper);
+  }
+
+  @PostMapping(value = TYPES + "/{" + TYPE_ID + "}")
+  @ApiOperation(
+      value = "Create a type this enables row level secure an entity",
+      response = ResponseEntity.class)
+  public ResponseEntity enableRLS(
+      HttpServletRequest request, @PathVariable(value = TYPE_ID) String typeId)
+      throws URISyntaxException {
+    permissionService.addType(typeId);
+    return ResponseEntity.created(new URI(request.getRequestURI())).build();
```

Comment:
Use ServletUriComponentsBuilder to get the 'user' request URI instead of a request URI that went from a proxy to the servlet container (see other code in code base).